react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+class Accordion extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { bodyVisible: false };
+    this.closeBody = this.closeBody.bind(this);
+    this.decipherID = this.decipherID.bind(this);
+  }
+
+  decipherID(event) {
+    for (let i = 0; i < this.props.topics.length; i++) {
+      if (this.props.topics[i].heading === event.target.textContent) {
+        this.setState({ currentID: this.props.topics[i].id });
+      }
+    }
+  }
+
+  closeBody() {
+    this.setState({ currentID: 0 });
+  }
```

Comment:
Rather than listen for double clicks, we want to close the currently open topic. So you will need to modify your `decipherID` method.

Try having a state like this

```js
this.state = {
  currentTopicID: null
};
```

If the same one gets clicked after it was already open, set `currentTopicID` back to `null`.